### PR TITLE
[Overflow-10] API Get Permissions of User

### PR DESCRIPTION
**Pull request by jules-lucero**
## Backlog Link
https://framgiaph.backlog.com/view/SUN_OVERFLOW-10

## Commands
```
cd backend
php artisan db:migrate
php artisan db:seed
```

## Pre-conditions
- Should setup environment variables for the Database

## Expected Output
- [x] You should be able to get all the permissions of a specific user

## Notes
- I also updated the User type

## Screenshots
image

### CODE REVIEW

**Review by IanUrriza10 — APPROVED** [review]

**Comment by IanUrriza10** [comment]
Reviewed: 20mins
Comments: 0
Iteration: 0

File: backend/graphql/user/query.graphql [hunk 1]
```diff
@@ -1,4 +1,10 @@
 extend type Query {
     user(id: ID @eq): User @find
     users: [User!]! @all
+
+    role(id: ID @eq): Role @find
+    roles: [Role!]! @all
+
+    permission(id: ID @eq): Permission @find
+    permissions: [Permission!]! @all
```
Comment (jescabillas) [inline backend/graphql/user/query.graphql:9]: Please make a separate folder for roles and include the permission there

File: backend/graphql/user/type.graphql [hunk 2]
```diff
@@ -1,7 +1,30 @@
...
+    permissions: [Permission!]! @belongsToMany
+}
+
+type Permission {
+    id: ID!
+    name: String!
+    description: String!
     created_at: String!
     updated_at: String
+    roles: [Role!]! @belongsToMany
```
Comment (jescabillas) [inline backend/graphql/user/type.graphql:29]: Please make a separate folder for roles and include the permission there

**Review by jescabillas — COMMENTED** [review]
Please make a mutation for creating a role since there's a feature to add roles

**Pull request merged**